code/modules/reagents/reagents.dm

```diff
@@ -5,7 +5,7 @@ GLOBAL_LIST_INIT(name2reagent, build_name2reagent())
 	for (var/t in subtypesof(/datum/reagent))
 		var/datum/reagent/R = t
 		if (length(initial(R.name)))
-			.[ckey(initial(R.name))] = t
+			LAZYADD(.[ckey(initial(R.name))], t)
```

Comment:
LAZYADD or LAZYSET?

Reply:
LAZYADD
. is an associated list, mapping the reagent name (the ckey bit) to a list of reagent datums. The datum list itself is just indexed.

Reply:
its lazyset, lazyadd is adding, before its easy to see it was just setting . not adding anything to it.

Reply:
Dont self resolve in future